### ISSUE TITLE
`treemacs-icons-dired-mode` variable overwritten when entering subdirectory

### ISSUE THREAD

**Issue by mcp292** [syl20bnr/spacemacs#15242]
#### Description :octocat:
`treemacs-icons-dired-mode` variable overwritten when entering subdirectory.

#### Reproduction guide :beetle:
When doing `SPC SPC customize-variable RET treemacs-icons-dired-mode` and setting to `nil` then hitting "Apply and Save", only the top level directory respects this setting.

*Observed behaviour:* :eyes: :broken_heart:
Once I enter a subdirectory, the setting is overwritten and `customize-variable` now reads "CHANGED outside of Customize". 

If I re-open spacemacs, the top level directory still respects this setting, and `customize-variable` reads `nil`.

This is what `customize-variable` put in my `.spacemacs`:
```lisp
(defun dotspacemacs/emacs-custom-settings ()
  "Emacs custom settings.
This is an auto-generated function, do not modify its content directly, use
Emacs customize menu instead.
This function is called at the very end of Spacemacs initialization."
(custom-set-variables
 ;; custom-set-variables was added by Custom.
 ;; If you edit it by hand, you could mess it up, so be careful.
 ;; Your init file should contain only one such instance.
 ;; If there is more than one, they won't work right.
 '(display-line-numbers t)
 '(evil-want-Y-yank-to-eol t)
 '(fill-column 72)
 '(magit-display-buffer-function 'magit-display-buffer-fullframe-status-v1)
 '(package-selected-packages
   '(treemacs-magit smeargle orgit-forge orgit org-rich-yank org-projectile org-category-capture org-present org-pomodoro alert log4e gntp org-mime org-download org-contrib org-cliplink org htmlize helm-org-rifle helm-git-grep gnuplot gitignore-templates git-timemachine git-modes git-messenger git-link forge yaml markdown-mode magit ghub closql emacsql-sqlite emacsql treepy magit-section git-commit with-editor transient flyspell-correct-helm flyspell-correct evil-org auto-dictionary ws-butler writeroom-mode visual-fill-column winum volatile-highlights vi-tilde-fringe uuidgen undo-tree treemacs-projectile treemacs-persp treemacs-icons-dired treemacs-evil treemacs cfrs pfuture posframe toc-org symon symbol-overlay string-inflection string-edit spaceline-all-the-icons memoize all-the-icons spaceline powerline restart-emacs request rainbow-delimiters quickrun popwin persp-mode password-generator paradox spinner overseer org-superstar open-junk-file nameless multi-line shut-up macrostep lorem-ipsum link-hint inspector info+ indent-guide hungry-delete hl-todo highlight-parentheses highlight-numbers parent-mode highlight-indentation helm-xref helm-themes helm-swoop helm-purpose window-purpose imenu-list helm-projectile helm-org helm-mode-manager helm-make helm-ls-git helm-flx helm-descbinds helm-ag google-translate golden-ratio flycheck-package package-lint flycheck pkg-info epl flycheck-elsa flx-ido flx fancy-battery eyebrowse expand-region evil-visualstar evil-visual-mark-mode evil-unimpaired f evil-tutor evil-textobj-line evil-surround evil-numbers evil-nerd-commenter evil-mc evil-matchit evil-lisp-state evil-lion evil-indent-plus evil-iedit-state evil-goggles evil-exchange evil-escape evil-ediff evil-easymotion evil-collection annalist evil-cleverparens smartparens evil-args evil-anzu anzu eval-sexp-fu emr iedit clang-format projectile paredit list-utils elisp-slime-nav editorconfig dumb-jump s drag-stuff dired-quick-sort devdocs define-word column-enforce-mode clean-aindent-mode centered-cursor-mode auto-highlight-symbol ht dash auto-compile packed aggressive-indent ace-window ace-link ace-jump-helm-line helm avy helm-core popup which-key use-package pcre2el hydra lv hybrid-mode font-lock+ evil goto-chg dotenv-mode diminish bind-map bind-key async))
 '(treemacs-icons-dired-mode nil nil (treemacs-icons-dired)))
(custom-set-faces
 ;; custom-set-faces was added by Custom.
 ;; If you edit it by hand, you could mess it up, so be careful.
 ;; Your init file should contain only one such instance.
 ;; If there is more than one, they won't work right.
 )
)
```

*Expected behaviour:* :heart: :smile:
All directories including subdirectories respect this setting.

#### System Info :computer:
- OS: gnu/linux
- Emacs: 27.2
- Spacemacs: 0.300.0
- Spacemacs branch: develop (rev. 0206197b2)
- Graphic display: t
- Distribution: spacemacs
- Editing style: vim
- Completion: helm
- Layers:
```elisp
(emacs-lisp git helm multiple-cursors org spell-checking treemacs)
```
- System configuration features: XPM JPEG TIFF GIF PNG RSVG CAIRO SOUND GPM DBUS GSETTINGS GLIB NOTIFY INOTIFY ACL GNUTLS LIBXML2 FREETYPE HARFBUZZ M17N_FLT LIBOTF ZLIB TOOLKIT_SCROLL_BARS GTK3 X11 XDBE XIM MODULES THREADS LIBSYSTEMD JSON PDUMPER LCMS2 GMP

**Comment by lebensterben** [comment]
https://www.gnu.org/software/emacs/manual/html_node/emacs/Changing-a-Variable.html
> The ‘[Apply]’ button applies the settings for the current session. The ‘[Apply and Save]’ button applies the settings and saves them for future sessions

So what's the unexpected behaviour here?

**Comment by mcp292** [comment]
Re-read.

**Comment by mcp292** [comment]
Re-open. What are you doing?

**Comment by lebensterben** [comment]
From your "reproduction guide", I see no unexpected behaviour.

After re-reading both that and Emacs manual, I conclude that this is not a bug.

**Comment by mcp292** [comment]
This is a lack of careful reading on your end. I'll entertain you. 

I hit `Apply and Save`, which as you note, applies settings and saves for future sessions. In any proceeding session, what's unexpected is the fact that in the top level directory viewed by `dired`, this variable takes effect and no icons are present. Good, that's what I want. 

The problem is, if you continue reading, as soon as I enter into a subdirectory on that page, the icons re-appear, ignoring the current variable setting and resetting the variable to `non-nil`. 

What part of this is expected?

**Comment by lebensterben** [comment]
It's saved for future sessions. Have you restarted Emacs? You didn't mention it so assume the answer is no.

**Comment by mcp292** [comment]
> It's saved for future sessions. Have you restarted Emacs? You didn't mention it so assume the answer is no.

This is incorrect. The message in the minibuffer when putting the cursor on `Apply and Save` reads:
> Apply settings and save for future sessions.

If that was your concern, ask. Don't close the issue. Why do I have to explain this to you? Your rudeness really deters contribution, and is unwelcome here. 

I did mention in my recent comment 
> In any proceeding session...

and in the original post which you failed to read carefully
> If I re-open spacemacs, the top level directory still respects this setting, and customize-variable reads nil.

So to answer your unrepresented question which I answered from the outset, yes I have restarted emacs after setting this variable. 

Will you collaborate now and stop being bitter?
